## Actividad Elaboracion de casos de COVID-19 en CDMX usando como punto de referencia las Estaciones del Metro.

### Objetivo de la actividad:
Como objetivo principal de la actividad se tiene buscar una relación entre los casos de COVID-19 y las estaciones del metro. Indentificando __puntos rojos__ en alguna colocia a la que pertenesca esa estación del metro. Por ejemplo


Si se encuentra que el metro _Hidalgo_ que esta en una __Colonia__ con demasiados casos de COVID-19 poder identificarlo por medio de un mapa desplegado en _R_ y poder observar los datos de esta colonia, como lo son su _Colonia_,_Alcaldia_ y _Numero de casos_.


### Resultado de la actividad.
Como resultado de esta actividad se pretende obtener los siguientes recursos:
* DataSet con información de los casos por colonia. Teniendo información como el numero de casos, nombre de la colonia, alcaldia, coordenadas geograficas.
* DataSet con informacion de las lineas del metro. Teniendo informacion como el nombre la estacion, linea y coordenadas geograficas.
* Mapa elaborado en una aplicacion Shiny Dashboard, con informacion de los DataSets previamente mencionados.


### Elaboración del primer resultado.

Para obtener el DataSet con el numero de casos por colonia, se descargo de la siguiente [liga](https://datos.cdmx.gob.mx/dataset/covid-19-sinave-ciudad-de-mexico-a-nivel-colonia/resource/ff1d4cbf-5985-45db-b40f-d820ce2b01a2). Y se inserto en la carpeta __docs__ , con el nombre 
`sinave_folios_colonias.csv`.


Y se le realizaron las siguientes operaciones.

In [2]:
#Importacion de librerias necesarias
import pandas as pd

In [3]:
data = pd.read_csv("docs/sinave_folios_colonias.csv")

Realizacion de analisis exploratorio

In [4]:
data.head()

colonia      alcaldia nueva_clave  total  cuartil  quintil  \
0                  ALDANA  AZCAPOTZALCO      02-002      8        1        2   
1           ANGEL ZIMBRON  AZCAPOTZALCO      02-005      6        1        1   
2                  ARENAL  AZCAPOTZALCO      02-006      9        2        2   
3  CENTRO DE AZCAPOTZALCO  AZCAPOTZALCO      02-007     11        2        3   
4                CLAVERIA  AZCAPOTZALCO      02-008      8        1        2   

   octil  decil                                          geo_shape  \
0      2      3  {"type": "Polygon", "coordinates": [[[-99.1485...   
1      1      1  {"type": "Polygon", "coordinates": [[[-99.1904...   
2      3      3  {"type": "Polygon", "coordinates": [[[-99.1518...   
3      4      5  {"type": "Polygon", "coordinates": [[[-99.1895...   
4      2      3  {"type": "Polygon", "coordinates": [[[-99.1874...   

                                           geo_point  
0  { "type": "Point", "coordinates": [-99.1499968...  
1  { "type": "Point", "coordinates": [-99.1891178...  
2  { "type": "Point", "coordinates": [-99.1520094...  
3  { "type": "Point", "coordinates": [-99.1863123...  
4  { "type": "Point", "coordinates": [-99.1818671...

In [5]:
data.tail()

colonia             alcaldia nueva_clave  total  cuartil  \
729  MOCTEZUMA 2A SECCION IV  VENUSTIANO CARRANZA      17-082     18        4   
730                MORELOS I  VENUSTIANO CARRANZA      17-083     15        3   
731               MORELOS II  VENUSTIANO CARRANZA      17-084     30        4   
732     PENSADOR MEXICANO II  VENUSTIANO CARRANZA      17-086     16        3   
733                      NaN                  NaN         NaN   2572        4   

     quintil  octil  decil                                          geo_shape  \
729        4      7      8  {"type": "Polygon", "coordinates": [[[-99.0996...   
730        4      6      7  {"type": "Polygon", "coordinates": [[[-99.1237...   
731        5      8     10  {"type": "Polygon", "coordinates": [[[-99.1212...   
732        4      6      8  {"type": "Polygon", "coordinates": [[[-99.0852...   
733        5      8     10                                                NaN   

                                             geo_point  
729  { "type": "Point", "coordinates": [-99.0977018...  
730  { "type": "Point", "coordinates": [-99.1230601...  
731  { "type": "Point", "coordinates": [-99.1183549...  
732  { "type": "Point", "coordinates": [-99.0885066...  
733                                                NaN

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 734 entries, 0 to 733
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   colonia      733 non-null    object
 1   alcaldia     733 non-null    object
 2   nueva_clave  733 non-null    object
 3   total        734 non-null    int64 
 4   cuartil      734 non-null    int64 
 5   quintil      734 non-null    int64 
 6   octil        734 non-null    int64 
 7   decil        734 non-null    int64 
 8   geo_shape    732 non-null    object
 9   geo_point    733 non-null    object
dtypes: int64(5), object(5)
memory usage: 57.5+ KB


In [9]:
data.columns

Index(['colonia', 'alcaldia', 'nueva_clave', 'total', 'cuartil', 'quintil',
       'octil', 'decil', 'geo_shape', 'geo_point'],
      dtype='object')

In [10]:
data.shape

(734, 10)

In [11]:
# Revisar datos nulos 

data.isnull().sum()

colonia        1
alcaldia       1
nueva_clave    1
total          0
cuartil        0
quintil        0
octil          0
decil          0
geo_shape      2
geo_point      1
dtype: int64

In [17]:
# Revisar filas nulas
data[data.isnull().any(1)]

colonia    alcaldia nueva_clave  total  cuartil  \
403  SAN PABLO OZTOTEPEC (PBLO)  MILPA ALTA      09-008     94        4   
733                         NaN         NaN         NaN   2572        4   

     quintil  octil  decil geo_shape  \
403        5      8     10       NaN   
733        5      8     10       NaN   

                                             geo_point  
403  { "type": "Point", "coordinates": [-99.0724622...  
733                                                NaN

### Resultado del analisis exploratorio.

Como resultado se obtuvo los siguientes resultados:
* El DataSet inicial cuenta con 734 colonias y cuenta con 10 campos. Los cuales son: colonia, alcaldia, nueva_clave, total, cuartil, quintil,octil, decil, geo_shape y geo_point.
* Se encontro que hay 734 casos no asignados a ninguna colonia.
* Existencia de un campo llamado "geo_point", que cuenta con los datos de longitud y latitud
* Esistencia de datos nulos

### Acciones a tomar

Se tomara la opcion de eliminar los registros que no se identificaron y ademas los campos que no aportan nada al resultado que se planea obetenr. Ademas de separa el campo "geo_point" en dos campos denominados: _latitude_ y _longitude_. Por ultimo se renombrara las columnas.


Realizacion de operaciones:

In [19]:
# Eliminamos ultimo registro
data_1 = data.drop([733],axis=0).copy()

# Mostrar resultado
data_1.tail()

colonia             alcaldia nueva_clave  total  \
728  MOCTEZUMA 2A SECCION III  VENUSTIANO CARRANZA      17-081     14   
729   MOCTEZUMA 2A SECCION IV  VENUSTIANO CARRANZA      17-082     18   
730                 MORELOS I  VENUSTIANO CARRANZA      17-083     15   
731                MORELOS II  VENUSTIANO CARRANZA      17-084     30   
732      PENSADOR MEXICANO II  VENUSTIANO CARRANZA      17-086     16   

     cuartil  quintil  octil  decil  \
728        3        4      6      7   
729        4        4      7      8   
730        3        4      6      7   
731        4        5      8     10   
732        3        4      6      8   

                                             geo_shape  \
728  {"type": "Polygon", "coordinates": [[[-99.1001...   
729  {"type": "Polygon", "coordinates": [[[-99.0996...   
730  {"type": "Polygon", "coordinates": [[[-99.1237...   
731  {"type": "Polygon", "coordinates": [[[-99.1212...   
732  {"type": "Polygon", "coordinates": [[[-99.0852...   

                                             geo_point  
728  { "type": "Point", "coordinates": [-99.1026514...  
729  { "type": "Point", "coordinates": [-99.0977018...  
730  { "type": "Point", "coordinates": [-99.1230601...  
731  { "type": "Point", "coordinates": [-99.1183549...  
732  { "type": "Point", "coordinates": [-99.0885066...

In [28]:
# Eliminacion de las campo que no aportan informacion
data_2 = data_1.drop(['nueva_clave', 'cuartil', 'quintil','octil', 'decil', 'geo_shape'],axis=1).copy()

#Mostrar resutados
data_2.head()

colonia      alcaldia  total  \
0                  ALDANA  AZCAPOTZALCO      8   
1           ANGEL ZIMBRON  AZCAPOTZALCO      6   
2                  ARENAL  AZCAPOTZALCO      9   
3  CENTRO DE AZCAPOTZALCO  AZCAPOTZALCO     11   
4                CLAVERIA  AZCAPOTZALCO      8   

                                           geo_point  
0  { "type": "Point", "coordinates": [-99.1499968...  
1  { "type": "Point", "coordinates": [-99.1891178...  
2  { "type": "Point", "coordinates": [-99.1520094...  
3  { "type": "Point", "coordinates": [-99.1863123...  
4  { "type": "Point", "coordinates": [-99.1818671...

In [32]:
# Limpieza del campo "geo_point"

data_2['geo_point']=data_2['geo_point'].str.replace('"type": "Point", "coordinates":','')
data_2['geo_point']=data_2['geo_point'].str.replace('[','') 
data_2['geo_point']=data_2['geo_point'].str.replace('{','') 
data_2['geo_point']=data_2['geo_point'].str.replace(']','') 
data_2['geo_point']=data_2['geo_point'].str.replace('}','') 

In [35]:
# Separacion de latitud y longitud
data_2[['longitude','latitude']] = data_2['geo_point'].str.split(",",expand=True)

# Eliminacion de columna "geo_point_2d"
data_3 = data_2.drop(['geo_point'],axis=1).copy()

# Mostrar resultados 
data_3.head()

colonia      alcaldia  total         longitude  \
0                  ALDANA  AZCAPOTZALCO      8    -99.1499968041   
1           ANGEL ZIMBRON  AZCAPOTZALCO      6    -99.1891178561   
2                  ARENAL  AZCAPOTZALCO      9    -99.1520094444   
3  CENTRO DE AZCAPOTZALCO  AZCAPOTZALCO     11    -99.1863123775   
4                CLAVERIA  AZCAPOTZALCO      8    -99.1818671607   

         latitude  
0  19.4707366167   
1  19.4677940841   
2  19.4652942835   
3  19.4811765098   
4  19.4660515311

In [36]:
# Renombramiento de columnas

#Diccionario con nuevo nombres
diccionario_cabeceras = {
    'alcandia':'alcaldia',
    'colonia':'colonia',
    'total':'casos_totales',
    'latitude':'latitude',
    'longitude':'longitude'
}

#Aplicaion del diccionario
data_4 = data_3.rename(columns=diccionario_cabeceras)

# Mostrar resultados 
data_4.head()

colonia      alcaldia  casos_totales         longitude  \
0                  ALDANA  AZCAPOTZALCO              8    -99.1499968041   
1           ANGEL ZIMBRON  AZCAPOTZALCO              6    -99.1891178561   
2                  ARENAL  AZCAPOTZALCO              9    -99.1520094444   
3  CENTRO DE AZCAPOTZALCO  AZCAPOTZALCO             11    -99.1863123775   
4                CLAVERIA  AZCAPOTZALCO              8    -99.1818671607   

         latitude  
0  19.4707366167   
1  19.4677940841   
2  19.4652942835   
3  19.4811765098   
4  19.4660515311

### Se procede a guardar resultados

In [38]:
data_5.to_csv('data_sets/Alcaldias_data_set.csv',index=False)

### Elaboración del segundo resultado.

Para obtener el DataSet con las estaciones del metro, se descargo de la siguiente [liga](https://www.datos.cdmx.gob.mx/dataset/estaciones-del-metro/resource/d5b2001c-7978-4541-9e76-2c863b01d451?view_id=1907db1d-eae3-421d-9703-f6b9b96215dc). Y se inserto en la carpeta __docs__ , con el nombre 
`estaciones-del-metro.csv`.


Y se le realizaron las siguientes operaciones.

In [39]:
data_metro = pd.read_csv("docs/estaciones-del-metro.csv")

In [40]:
data_metro.head()

id                  geo_point_2d  \
0   0  19.3757967002,-99.1877081963   
1   1  19.3705202583,-99.1648166498   
2   2  19.3614164661,-99.1516874947   
3   3  19.3575230891,-99.1214990292   
4   4  19.3377244141,-99.1090561035   

                                           geo_shape           name  \
0  {"type": "Point", "coordinates": [-99.18770819...        Mixcoac   
1  {"type": "Point", "coordinates": [-99.16481664...         Zapata   
2  {"type": "Point", "coordinates": [-99.15168749...    Eje Central   
3  {"type": "Point", "coordinates": [-99.12149902...  Mexicaltzingo   
4  {"type": "Point", "coordinates": [-99.10905610...      Culhuacán   

  descriptio        lon        lat                   linea_esta  
0   Linea 12 -99.187708  19.375797             Linea 12-Mixcoac  
1   Linea 12 -99.164817  19.370520              Linea 12-Zapata  
2   Linea 12 -99.151687  19.361416         Linea 12-Eje Central  
3   Linea 12 -99.121499  19.357523       Linea 12-Mexicaltzingo  
4   Linea 12 -99.109056  19.337724  Linea 12-Pueblo Culhuacácan

In [41]:
data_metro.tail()

id                  geo_point_2d  \
191  191  19.4909502302,-99.0468935014   
192  192  19.4510576348,-99.0795128481   
193  193  19.4365072738,-99.1039374804   
194  194   19.442497616,-99.1235866762   
195  195  19.4464014494,-99.1531622142   

                                             geo_shape                  name  \
191  {"type": "Point", "coordinates": [-99.04689350...   Río de los Remedios   
192  {"type": "Point", "coordinates": [-99.07951284...     Deportivo Oceanía   
193  {"type": "Point", "coordinates": [-99.10393748...  Ricardo Flores Magón   
194  {"type": "Point", "coordinates": [-99.12358667...                Tepito   
195  {"type": "Point", "coordinates": [-99.15316221...            Buenavista   

    descriptio        lon        lat                    linea_esta  
191    Linea B -99.046894  19.490950   Linea B-Río de los Remedios  
192    Linea B -99.079513  19.451058     Linea B-Deportivo Oceanía  
193    Linea B -99.103937  19.436507  Linea B-Ricardo Flores Magón  
194    Linea B -99.123587  19.442498                Linea B-Tepito  
195    Linea B -99.153162  19.446401            Linea B-Buenavista

In [42]:
data_metro.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 196 entries, 0 to 195
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id            196 non-null    int64  
 1   geo_point_2d  196 non-null    object 
 2   geo_shape     196 non-null    object 
 3   name          196 non-null    object 
 4   descriptio    196 non-null    object 
 5   lon           196 non-null    float64
 6   lat           196 non-null    float64
 7   linea_esta    196 non-null    object 
dtypes: float64(2), int64(1), object(5)
memory usage: 12.4+ KB


In [44]:
data_metro.columns

Index(['id', 'geo_point_2d', 'geo_shape', 'name', 'descriptio', 'lon', 'lat',
       'linea_esta'],
      dtype='object')

In [46]:
data_metro.columns

Index(['id', 'geo_point_2d', 'geo_shape', 'name', 'descriptio', 'lon', 'lat',
       'linea_esta'],
      dtype='object')

In [47]:
data_metro.shape

(196, 8)

In [48]:
# Revisar datos nulos 

data_metro.isnull().sum()

id              0
geo_point_2d    0
geo_shape       0
name            0
descriptio      0
lon             0
lat             0
linea_esta      0
dtype: int64

### Resultado del analisis exploratorio.

Como resultado se obtuvo los siguientes resultados:
* El DataSet inicial cuenta con 196 estaciones del metro y cuenta con 8 campos. Los cuales son: id, geo_point_2d, geo_shape, name, descriptio, lon, lat y linea_esta.
* No hay presente ningun data nulo

### Acciones a tomar

Se eliminaran los campos id, geo_shape, name, descriptio, lon, lat y linea_esta. Ademas de separa el campo "geo_point_2d" en dos campos denominados: _latitude_ y _longitude_. Por ultimo se renombrara las columnas.


Realizacion de operaciones:

In [50]:
# Eliminacion de las campo que no aportan informacion
data_metro_1 = data_metro.drop(['id', 'geo_shape', 'lon', 'lat', 'linea_esta'],axis=1).copy()

#Mostrar resutados
data_metro_1.head()

geo_point_2d           name descriptio
0  19.3757967002,-99.1877081963        Mixcoac   Linea 12
1  19.3705202583,-99.1648166498         Zapata   Linea 12
2  19.3614164661,-99.1516874947    Eje Central   Linea 12
3  19.3575230891,-99.1214990292  Mexicaltzingo   Linea 12
4  19.3377244141,-99.1090561035      Culhuacán   Linea 12

In [53]:
# Separacion de latitud y longitud
data_metro_1[['latitude','longitude']] = data_metro_1['geo_point_2d'].str.split(",",expand=True)

# Eliminacion de columna "geo_point_2d"
data_metro_2 = data_metro_1.drop(['geo_point_2d'],axis=1).copy()

# Mostrar resultados 
data_metro_2.head()

name descriptio       latitude       longitude
0        Mixcoac   Linea 12  19.3757967002  -99.1877081963
1         Zapata   Linea 12  19.3705202583  -99.1648166498
2    Eje Central   Linea 12  19.3614164661  -99.1516874947
3  Mexicaltzingo   Linea 12  19.3575230891  -99.1214990292
4      Culhuacán   Linea 12  19.3377244141  -99.1090561035

In [54]:
# Renombramiento de columnas

#Diccionario con nuevo nombres
diccionario_cabeceras_metro = {
    'name':'nombre_estacion',
    'descriptio':'estacion',
    'latitude':'latitude',
    'longitude':'longitude'
}

#Aplicaion del diccionario
data_metro_3 = data_metro_2.rename(columns=diccionario_cabeceras_metro)

# Mostrar resultados 
data_metro_3.head()

nombre_estacion  estacion       latitude       longitude
0         Mixcoac  Linea 12  19.3757967002  -99.1877081963
1          Zapata  Linea 12  19.3705202583  -99.1648166498
2     Eje Central  Linea 12  19.3614164661  -99.1516874947
3   Mexicaltzingo  Linea 12  19.3575230891  -99.1214990292
4       Culhuacán  Linea 12  19.3377244141  -99.1090561035

### Se procede a guardar resultados

In [58]:
data_metro_3.to_csv('data_sets/Estaciones_metro.csv',index=False, encoding='latin1')